pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,445 @@
+    display_format = Instrument.control(
+        ":CALC1:FORM?",":CALC1:FORM %s",
+        """ A property that allow user to set the measurement parameters to be displayed and output as DATA1. """,
+        validator=strict_discrete_set,
+        values={"Real": "REAL",
+                "Mlinear": "MLIN",
+                "Imaginary": "IMAG",
+                "Phase": "PHAS",
+                "Noise": "NOIS",
+                "Aux1": "AUX1",
+                "Real2": "REAL2",
+                "Mlinear2": "MLIN2"},
+        map_values=True
+    )
+
+    reference_current = Instrument.control(
+        ":CALC1:MATH:CURR?",":CALC1:MATH:CURR %g",
+        """ A property that allow user to set the current reference value for normalize calculation. """,
+        validator=truncated_range,
+        values=[1e-15, 1e-6]
+    )
+
+    calculation_format = Instrument.control(
+        ":CALC1:MATH:EXPR:NAME?", ":CALC1:MATH:EXPR:NAME %s", 
+        """ A property that allow user to normalize calculation format. """,
+        validator=strict_discrete_set,
+        values={"DB": "DB, 'dB'",
+                "Reference": "PCNT, ''",
+                "Sensitivity": "PCFS, ''",
+                },
+        map_values=True
+    )
+
+    reference_voltage = Instrument.control(
+        ":CALC1:MATH:VOLT?", ":CALC1:MATH:VOLT %g",
+        """ A property that allow user to set the voltage reference value for normalize calculation. """,
+        validator=truncated_range,
+        values=[1e-9, 1e+1]
+    )
+
+    offset_enable = Instrument.control(
+        ":CALC1:OFFS:STAT?", ":CALC1:OFFS:STAT %s",
+        """ A property that allow user to enable/disable offset for the primary PSD's X output. """,
+        validator=strict_discrete_set,
+        values={"On": 1, "Off": 0},
+        map_values=True
+    )
+
+    primary_offset = Instrument.control(
+        "CALC1:OFFS?",":CALC1:OFFS:STAT ON; :CALC1:OFFS %g",
+        """ A property that allow user to offset for the primary PSD's X output. """,
+        validator=truncated_range,
+        values=[-105, 105]
+    )
+
+    calculation_method = Instrument.control(
+        ":CALC5:MATH?", ":CALC5:MATH %s",
+        """ A property that allow user to set the calculation method for measurement value to be displayed and output. """,
+        validator=strict_discrete_set,
+        values={"Off": "OFF",
+                "Expand": "EXP",
+                "Normalize": "NORM",
+                "Ratio": "RAT"},
+        map_values=True
+    )
+
+    fetch_data = Instrument.measurement(
+        ":FETC?",
+        """ Queries the most recent measurement data. """
+    )
+
+    #############
+    # Frequency #
+    #############
+    frequency = Instrument.measurement(
+        ":FREQ?",
+        """ Return the fundamental wave, primary frequency. """
+    )
+
+    harmonics_pre_enable = Instrument.control(
+        ":FREQ:HARM?", ":FREQ:HARM %d",
+        """ A property that enable/disable the harmonics measurement. """,
```

Comment:
It would be nice to (explicitely) mention the possible values in the docstring. You could also consider using booleans rather than the "On" and "Off" strings.